data-prepper-plugins/service-map-stateful/src/main/java/com/amazon/dataprepper/plugins/prepper/ServiceMapStatefulPrepper.java

```diff
@@ -127,41 +128,37 @@ private static File createPath(File path) {
      */
     @Override
-    public Collection<Record<String>> doExecute(Collection<Record<ExportTraceServiceRequest>> records) {
-        final Collection<Record<String>> relationships = windowDurationHasPassed() ? evaluateEdges() : EMPTY_COLLECTION;
+    public Collection<Record<Event>> doExecute(Collection<Record<Span>> records) {
+        final Collection<Record<Event>> relationships = windowDurationHasPassed() ? evaluateEdges() : EMPTY_COLLECTION;
         final Map<byte[], ServiceMapStateData> batchStateData = new TreeMap<>(SignedBytes.lexicographicalComparator());
-        records.forEach(i -> i.getData().getResourceSpansList().forEach(resourceSpans -> {
-            OTelHelper.getServiceName(resourceSpans.getResource()).ifPresent(serviceName -> resourceSpans.getInstrumentationLibrarySpansList().forEach(
-                    instrumentationLibrarySpans -> {
-                        instrumentationLibrarySpans.getSpansList().forEach(
-                                span -> {
-                                    if (OTelHelper.checkValidSpan(span)) {
-                                        try {
-                                            batchStateData.put(
-                                                    span.getSpanId().toByteArray(),
-                                                    new ServiceMapStateData(
-                                                            serviceName,
-                                                            span.getParentSpanId().isEmpty() ? null : span.getParentSpanId().toByteArray(),
-                                                            span.getTraceId().toByteArray(),
-                                                            span.getKind().name(),
-                                                            span.getName()));
-                                        } catch (RuntimeException e) {
-                                            LOG.error("Caught exception trying to put service map state data into batch", e);
-                                        }
-                                        if (span.getParentSpanId().isEmpty()) {
-                                            try {
-                                                currentTraceGroupWindow.put(span.getTraceId().toByteArray(), span.getName());
-                                            } catch (RuntimeException e) {
-                                                LOG.error("Caught exception trying to put trace group name", e);
-                                            }
-                                        }
-                                    } else {
-                                        LOG.warn("Invalid span received");
-                                    }
-                                });
+        records.forEach(record -> {
+            final Span span = record.getData();
+            if (span.getServiceName() != null) {
```

Comment:
How should we handle span's with a null value for a `ServiceName`? At the moment it looks like we are just dropping them.

Given that this mimics the existing flow let's capture a follow up task to address this.

Reply:
Also, we should coordinate with [trace-analytics UI](https://github.com/opensearch-project/observability) for planning the change.

Reply:
Good call out. Let's make a follow up issue. We can decide there if and how we may want to address this.